include/oneapi/dpl/async

```diff
@@ -19,13 +19,25 @@
 #include "oneapi/dpl/internal/common_config.h"
 #include "oneapi/dpl/pstl/onedpl_config.h"
 
-#include "oneapi/dpl/internal/async_extension_defs.h"
+#if _ONEDPL_BACKEND_SYCL
+#    if !_ONEDPL_ASYNC_FORWARD_DECLARED
+#        include "oneapi/dpl/internal/async_extension_defs.h"
+#        define _ONEDPL_ASYNC_FORWARD_DECLARED 1
+#    endif // !_ONEDPL_ASYNC_FORWARD_DECLARED
 
-#if _ONEDPL_EXECUTION_POLICIES_DEFINED
-// If <execution> has already been included, pull in implementations
-#    include "oneapi/dpl/internal/async_impl/async_impl.h"
-#    include "oneapi/dpl/internal/async_impl/glue_async_impl.h"
-#endif // _ONEDPL_EXECUTION_POLICIES_DEFINED
+#    if _ONEDPL_EXECUTION_POLICIES_DEFINED
+         // If <execution> has already been included, pull in implementations
+#        include "oneapi/dpl/internal/async_impl/async_impl.h"
```

Comment:
Actually, should `async_impl.h` be included directly into the public header? If it only contains internal implementation details and no public API definitions, then it should be included into `glue_async_impl.h` but not here.

Reply:
I see that it does include some public API. But then, why do we have two distinct implementation headers? Should we move everything into `glue_async_impl.h` (and remove any possible duplication)?

@MikeDvorskiy do you know what's the difference between the definitions in `async_impl.h` and `glue_async_impl.h`?

Reply:
Fixed: inclusion of `async_impl.h` removed from public header `async'.

Reply:
Fixed: ```async_impl.h``` was removed.